src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php

```diff
@@ -0,0 +1,148 @@
+ * @author James Watson <james@sitepulse.org>
+ */
+final class JUnitOutputPrinter extends StreamOutputPrinter
+{
+    const XML_VERSION  = '1.0';
+    const XML_ENCODING = 'UTF-8';
+
+    /**
+     * @var \DOMDocument
+     */
+    private $domDocument;
+    /**
+     * @var \DOMElement
+     */
+    private $currentTestsuite;
+    /**
+     * @var \DOMElement
+     */
+    private $currentTestcase;
+    /**
+     * @var \DOMElement
+     */
+    private $testSuites;
+
+    public function __construct(FilesystemOutputFactory $outputFactory)
+    {
+        parent::__construct($outputFactory);
+    }
+
+    /**
+     * Creates a new JUnit file.
+     *
+     * The file will be initialized with an XML definition and the root element.
+     *
+     * @param string $name                 The filename (without extension) and default value of the name attribute
+     * @param array  $testsuitesAttributes Attributes for the root element
+     */
+    public function createNewFile($name, array $testsuitesAttributes = array())
+    {
+        $this->setFileName(strtolower(trim(preg_replace('/[^[:alnum:]_]+/', '_', $name), '_')));
+
+        $this->domDocument = new \DOMDocument(self::XML_VERSION, self::XML_ENCODING);
+        $this->domDocument->formatOutput = true;
+
+        $this->testSuites = $this->domDocument->createElement('testsuites');
+        $this->domDocument->appendChild($this->testSuites);
+        $this->addAttributesToNode($this->testSuites, array_merge(array('name' => $name), $testsuitesAttributes));
+        $this->flush();
+    }
+
+    /**
+     * Adds a new <testsuite> node.
+     *
+     * @param array $testsuiteAttributes
+     */
+    public function addTestsuite(array $testsuiteAttributes = array())
+    {
+        $this->currentTestsuite = $this->domDocument->createElement('testsuite');
+        $this->testSuites->appendChild($this->currentTestsuite);
+        $this->addAttributesToNode($this->currentTestsuite, $testsuiteAttributes);
+    }
+
+
+    /**
+     * Adds a new <testcase> node.
+     *
+     * @param array $testcaseAttributes
+     */
+    public function addTestcase(array $testcaseAttributes = array())
+    {
+        $this->currentTestcase = $this->domDocument->createElement('testcase');
+        $this->currentTestsuite->appendChild($this->currentTestcase);
+        $this->addAttributesToNode($this->currentTestcase, $testcaseAttributes);
+    }
+
+    /**
+     * Add a testcase child element.
+     *
+     * @param string $nodeName
+     * @param array  $nodeAttributes
+     * @param string $nodeValue
+     */
+    public function addTestcaseChild($nodeName, array $nodeAttributes = array(), $nodeValue = null)
+    {
+        $childNode = $this->domDocument->createElement($nodeName, $nodeValue);
+        $this->currentTestcase->appendChild($childNode);
+        $this->addAttributesToNode($childNode, $nodeAttributes);
+    }
+
+    public function addAttributesToNode(\DOMElement $node, $attributes)
```

Comment:
the second argument should be typehinted as array.

And could this method be private ?